google/cloud/bigquery/table.py

```diff
@@ -2648,31 +2648,25 @@ def to_dataframe(
                 if pyarrow.types.is_timestamp(col.type)
             )
 
-        if len(record_batch) > 0:
-            df = record_batch.to_pandas(
+        df = record_batch.to_pandas(
+            date_as_object=date_as_object,
+            timestamp_as_object=timestamp_as_object,
+            integer_object_nulls=True,
+            types_mapper=_pandas_helpers.default_types_mapper(
                 date_as_object=date_as_object,
-                timestamp_as_object=timestamp_as_object,
-                integer_object_nulls=True,
-                types_mapper=_pandas_helpers.default_types_mapper(
-                    date_as_object=date_as_object,
-                    bool_dtype=bool_dtype,
-                    int_dtype=int_dtype,
-                    float_dtype=float_dtype,
-                    string_dtype=string_dtype,
-                    date_dtype=date_dtype,
-                    datetime_dtype=datetime_dtype,
-                    time_dtype=time_dtype,
-                    timestamp_dtype=timestamp_dtype,
-                    range_date_dtype=range_date_dtype,
-                    range_datetime_dtype=range_datetime_dtype,
-                    range_timestamp_dtype=range_timestamp_dtype,
-                ),
-            )
-        else:
-            # Avoid "ValueError: need at least one array to concatenate" on
-            # older versions of pandas when converting empty RecordBatch to
-            # DataFrame. See: https://github.com/pandas-dev/pandas/issues/41241
```

Comment:
https://github.com/pandas-dev/pandas/pull/41052 has released with pandas 1.3.0 (as mentioned in above issue: https://github.com/pandas-dev/pandas/issues/41241#issuecomment-856065757)

So if we can increase minimum pandas version to 1.3.0, we don't need this workaround

Reply:
We should really come up with a consistent policy across all Google packages:

* pandas-gbq supports 1.1.4+ (https://github.com/googleapis/python-bigquery-pandas/blob/b3a9af27960c6e2736835d6a94e116f550f94114/setup.py#L27)
* db-dtypes supports 1.2.0+ (https://github.com/googleapis/python-db-dtypes-pandas/blob/44a5aa24ef3219849e9e0bedeeaebc4b9ee53d9e/setup.py#L34)
* bigframes supports 1.5.3+ (https://github.com/googleapis/python-bigquery-dataframes/blob/7d00be67cf50fdf713c40912f207d14f0f65538f/setup.py#L53)
* Colab and BigQuery Studio Python notebooks are both in the 2.x range now.

CC @chalmerlowe @Linchin @shollyman @parthea

Reply:
Indeed, moving to a monorepo might help make the dependencies more consistent, ideally we might want any overlapped dependencies among the handwritten BigQuery projects to be consistent. 

But just for the purpose of this PR, I think it's reasonable to increase the minimum pandas version to 1.3.0, as long as it's not breaking anything.